src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurDetailLineGroupForCollector.java

```diff
@@ -319,33 +320,87 @@ public void buildAndAddOriginEntries(Consumer<OriginEntryFull> entryConsumer) {
                 ConcurSAECollectorLineType.ATM_UNUSED_CASH_ADVANCE)) {
-            addOffsetOriginEntriesForUnusedAtmCashAdvanceAmountLines(entryConsumer, unusedAtmAmountSubGroup);
+            addOffsetOriginEntriesForUnusedAtmCashAdvanceAmountLines(temporaryCollectorBatch::addOriginEntry, unusedAtmAmountSubGroup);
         }
         
-        addOriginEntriesForCorpCardPersonalExpenseDebitLines(entryConsumer);
-        addOriginEntriesForCorpCardPersonalExpenseCreditLines(entryConsumer);
+        addOriginEntriesForCorpCardPersonalExpenseDebitLines(temporaryCorporateCardPersonalExpenseCollectorBatch::addOriginEntry);
+        addOriginEntriesForCorpCardPersonalExpenseCreditLines(temporaryCollectorBatch::addOriginEntry);
+        
+        cleanupCashAndAddCorporateCardPersonalExpenseEntries(temporaryCollectorBatch, temporaryCorporateCardPersonalExpenseCollectorBatch);
+        
+        addEntries(entryConsumer, temporaryCollectorBatch.getOriginEntries());
+    }
+    
+	private void cleanupCashAndAddCorporateCardPersonalExpenseEntries(CollectorBatch temporaryCollectorBatch,
+			CollectorBatch temporaryCorporateCardPersonalExpenseCollectorBatch) {
+
+		List<OriginEntryFull> paymentOffsetCashEntries = temporaryCollectorBatch.getOriginEntries().stream().filter(
+				entry -> collectorHelper.paymentOffsetObjectCode.equalsIgnoreCase(entry.getFinancialObjectCode()))
+				.collect(Collectors.toList());
+		KualiDecimal cashTotal = calculateTotalAmountForEntries(paymentOffsetCashEntries);
+
+		List<OriginEntryFull> paymentOffsetCorporateCardPersonalExpenseDebitEntries = temporaryCorporateCardPersonalExpenseCollectorBatch
+				.getOriginEntries().stream().filter(entry -> collectorHelper.paymentOffsetObjectCode.equalsIgnoreCase(entry.getFinancialObjectCode()))
+				.collect(Collectors.toList());
+		List<OriginEntryFull> nonPaymentOffsetCorporateCardPersonalExpenseDebitEntries = temporaryCorporateCardPersonalExpenseCollectorBatch
+				.getOriginEntries().stream().filter(entry -> !collectorHelper.paymentOffsetObjectCode.equalsIgnoreCase(entry.getFinancialObjectCode()))
+				.collect(Collectors.toList());
+		
+		KualiDecimal corporateCardPersonalExpenseDebitTotal = calculateTotalAmountForEntries(
+				paymentOffsetCorporateCardPersonalExpenseDebitEntries);
+
+		if (cashTotal.isGreaterThan(corporateCardPersonalExpenseDebitTotal)) {
+			KualiDecimal totalDeductionsLeft = corporateCardPersonalExpenseDebitTotal;
+			for (OriginEntryFull entry : paymentOffsetCashEntries) {
+				KualiDecimal newTransactionAmount = entry.getTransactionLedgerEntryAmount().subtract(totalDeductionsLeft);
+				if (newTransactionAmount.isNegative()) {
+					totalDeductionsLeft = newTransactionAmount;
+					newTransactionAmount = KualiDecimal.ZERO;
+				} else {
+					totalDeductionsLeft = KualiDecimal.ZERO;
+				}
+				entry.setTransactionLedgerEntryAmount(newTransactionAmount);

```

Comment:
If an origin entry ends up with an amount of zero after adjustment, then it should be removed.

Reply:
added code to remove entries with zero amount